### PR TITLE
fix(artifacts): Catch retryable request timeout when uploading artifacts to AWS

### PR DESCRIPTION
**Pull request by nickpenaranda**
Fixes WB-11059
Fixes #NNNN

Description
-----------

Instead of the standard `408` status code, S3 will respond with [code `400` and an XML payload](https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html) containing an error string and detailed error message. This has been observed to happen intermittently for `PUT` (upload) requests, and more consistently when uploading larger artifacts.

This PR attempts to detect these out-of-band `RequestTimeout` errors and correctly retry the request.

Testing
-------
How was this PR tested? TBD

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/sdk/internal/internal_api.py [hunk 1]
```diff
@@ -1887,11 +1887,20 @@ def upload_file(
             response_content = e.response.content if e.response is not None else ""
             logger.error(f"upload_file response body: {response_content}")
             status_code = e.response.status_code if e.response is not None else 0
+            # S3 reports retryable request timeouts out-of-band
+            is_aws_retryable = (
+                status_code == 400 and "RequestTimeout" in response_content
+            )
```
Comment (dmitryduev) [inline wandb/sdk/internal/internal_api.py:1893]: This feels dangerous, let me see if we can find a more reliable way to detect this...

Comment (dmitryduev) [reply wandb/sdk/internal/internal_api.py:1893]: as discussed elsewhere, let's add checking `x-amz-meta-md5` and get this one merged.

Comment (nickpenaranda) [reply wandb/sdk/internal/internal_api.py]: done

**Review by dmitryduev — APPROVED** [review]
🙏🏻

**Pull request merged**